src/assets/wise5/authoringTool/peer-grouping/author-peer-grouping-dialog/author-peer-grouping-dialog.component.ts

```diff
@@ -39,10 +47,30 @@ export abstract class AuthorPeerGroupingDialogComponent implements OnInit {
   }
 
   protected updatePeerGroupingLogic(): void {
-    this.peerGrouping.logic =
-      this.logicType === DIFFERENT_IDEAS_VALUE
-        ? `${DIFFERENT_IDEAS_VALUE}("${this.referenceComponent.nodeId}", "${this.referenceComponent.componentId}")`
-        : this.logicType;
+    switch (this.logicType) {
+      case DIFFERENT_IDEAS_VALUE:
+        this.peerGrouping.logic = `${DIFFERENT_IDEAS_VALUE}("${this.referenceComponent.nodeId}", "${this.referenceComponent.componentId}")`;
+        break;
+      case DIFFERENT_SCORES_VALUE:
+        this.peerGrouping.logic = this.generateDifferentScoresLogic(
+          this.referenceComponent,
+          this.mode
+        );
+        break;
+      default:
+        this.peerGrouping.logic = this.logicType;
+    }
+  }
+
+  private generateDifferentScoresLogic(
+    referenceComponent: ReferenceComponent,
+    mode: string
+  ): string {
+    if (this.mode == null) {
```

Comment:
Since this func is called with this.mode as second param, we don't need to reference it again?
```suggestion
    if (mode == null) {
```